app/jobs/descriptive_metadata_import_job.rb

```diff
@@ -14,7 +16,14 @@ class DescriptiveMetadataImportJob < GenericJob
   # @option params [String] :use_barcodes_option option to update the barcodes
   def perform(bulk_action_id, params)
     super
-    csv = CSV.parse(params[:csv_file], headers: true)
+    # uploaded_file is the file provided by the user, it can be a CSV, ODS or Excel file (xls or xlsx)
+    uploaded_file = params[:uploaded_file]
+
+    raise 'Unsupported upload file type' unless %w[.csv .ods .xls .xlsx].include? File.extname(uploaded_file)
+
+    spreadsheet = Roo::Spreadsheet.open(uploaded_file) # open the spreadsheet
```

Comment:
I think roo should be in CsvUploadNormalizer.

Reply:
Ahh, I missed this class.  Though doing it here will then allow Excel files to be uploaded everywhere (and not just in the Desc Metadata job as specced in the ticket).  Though I supposed this won't matter much, since the user should be able to continue using CSV and letting roo deal with it?

Reply:
Exactly.

Reply:
Another issue is that we have argo on two different machines. You're implementation here would require that all the nodes have access to the same upload area. I'm not sure we can make that assumption.

Reply:
But using the `CsvUploadNormalized` class to just open with roo and convert to CSV immediately gets around this right?  Because that class is dealing with the file that the user just uploaded ?